include/okapi/impl/device/opticalSensor.hpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#pragma once
+
+#include "api.h"
+#include "okapi/api/control/controllerInput.hpp"
+#include "okapi/api/filter/passthroughFilter.hpp"
+#include <memory>
+
+namespace okapi {
+class OpticalSensor : public ControllerInput<double> {
+  public:
+  /**
+   * An optical sensor on a V5 port.
+   *
+   * ```cpp
+   * auto os = OpticalSensor(1);
+   * ```
+   *
+   * @param iport The V5 port the device uses
+   */
+  OpticalSensor(std::uint8_t iport);
+
+  virtual ~OpticalSensor();
+
+  /**
+   * Returns the current sensor's hue value (0-359.99).
```

Comment:
To better match okapi style, you might want to do:
```suggestion
   * Returns The current sensor's hue value in the range [0, 360).
```
Also, you could surround the range notation with double backticks, if it makes the documentation look nicer.